bot/glossary_command_test.go

```diff
@@ -0,0 +1,167 @@
+package bot
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"testing"
+
+	"github.com/quintilesims/slackbot/db"
+	"github.com/quintilesims/slackbot/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGlossaryDefine(t *testing.T) {
+	inputs := map[string]string{
+		"foo": "bar",
+		"bar": "baz is multiple words",
+		"baz": "foo",
+	}
+
+	store := newMemoryStore(t)
+	for key, val := range inputs {
+		w := bytes.NewBuffer(nil)
+		cmd := NewGlossaryCommand(store, w)
+		input := fmt.Sprintf("!glossary define %s %s", key, val)
+
+		if err := runTestApp(cmd, input); err != nil {
+			t.Fatal(err)
+		}
+
+		assert.Contains(t, w.String(), key)
+		assert.Contains(t, w.String(), val)
+	}
+
+	result := models.Glossary{}
+	if err := store.Read(db.GlossaryKey, &result); err != nil {
+		t.Fatal(err)
+	}
+
+	expected := models.Glossary{
+		"foo": "bar",
+		"bar": "baz is multiple words",
+		"baz": "foo",
+	}
+
+	assert.Equal(t, expected, result)
+}
+
+func TestGlossaryDefineErrors(t *testing.T) {
+	inputs := []string{
+		"!glossary define",
+		"!glossary define foo",
+	}
+
+	store := newMemoryStore(t)
+	for _, input := range inputs {
+		t.Run(input, func(t *testing.T) {
+			cmd := NewGlossaryCommand(store, ioutil.Discard)
+			if err := runTestApp(cmd, input); err == nil {
+				t.Fatal("Error was nil!")
+			}
+		})
+	}
+}
+
+func TestGlossaryRemove(t *testing.T) {
+	glossary := models.Glossary{
+		"foo": "bar",
```

Comment:
[nit] maybe I should have had the definitions just be empty strings - that helps convey that the definitions don't matter for this test.